File: Python/A/631A-interview.py
n = int(input())
a = list(map(int, input().split()))
b = list(map(int, input().split()))
max_sum = 0
for l in range(n):
    for r in range(l, n):
        sum_a = 0
        sum_b = 0
        for i in range(l, r + 1):
            sum_a |= a[i]
            sum_b |= b[i]
        max_sum = max(max_sum, sum_a + sum_b)
print(max_sum)

File: C++/A/631A-interview.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, max_sum = 0;
    cin >> n;
    vector<int> a(n), b(n);
    for (int i = 0; i < n; ++i)
        cin >> a[i];
    for (int i = 0; i < n; ++i)
        cin >> b[i];
    for (int l = 0; l < n; ++l)
    {
        for (int r = l; r < n; ++r)
        {
            int sum_a = 0, sum_b = 0;

            for (int i = l; i <= r; ++i)
            {
                sum_a |= a[i];
                sum_b |= b[i];
            }
            max_sum = max(max_sum, sum_a + sum_b);
        }
    }
    cout << max_sum;
}
